images/metering-ansible-operator/roles/meteringconfig/tasks/reconcile.yml

```diff
@@ -14,8 +14,16 @@
   vars:
     hiveStorageType: "{{ meteringconfig_spec_overrides | json_query('storage.hive.type') }}"
 
+- name: Validate fields when OCP features are disabled
+  assert:
+    that:
+      - meteringconfig_spec['reporting-operator'] | json_query('spec.config.prometheus.url') != ""
+    msg: "Invalid configuration for non-OCP distributions. You must set the reporting-operator.spec.config.prometheus.url."
+  when: meteringconfig_ocp_disabled
+
 - name: Configure TLS
   include_tasks: configure_tls.yml
+  when: not meteringconfig_ocp_disabled
```

Comment:
This needs to be removed otherwise TLS being enabled, then disabled won't work correctly.